frontend/src/views/pages/vehicleLists/UserVehicleLists.vue

```diff
@@ -0,0 +1,313 @@
+<template>
+  <div>
+    <CCard>
+      <CCardHeader>
+        <slot name="header">Table of custom vehicle lists</slot>
+      </CCardHeader>
+      <CCardBody>
+        <CRow class="d-flex justify-content-between pl-3 pr-3">
+          <CButton color="primary mb-3" @click="addingVehicleList = true">Create a custom vehicle list</CButton>
+          <CRow class="m-0">
+            <CButton color="success mb-3 mr-2" @click="applyAction" v-if="selectedBulkAction != 'Bulk Action'">Apply</CButton>
+            <CButton color="secondary mb-3 mr-2" @click="cancelAction" v-if="selectedBulkAction != 'Bulk Action'">Cancel</CButton>
+            <CSelect 
+              :options="bulkActions"
+              :value.sync="selectedBulkAction"
+              :disabled="selectedBulkAction != 'Bulk Action'"
+            />
+          </CRow>
+        </CRow>
+        <CDataTable
+          :fields="tableFields"
+          :items="tableItems"
+          :sorter="true"
+          :hover="true"
+          :clickable-rows="true"
+          :column-filter="true"
+          @row-clicked="clickRow"
+          v-if="selectedBulkAction == 'Bulk Action'"
+        >
+          <template #date_created="{item}">
+            <td>
+              {{ getFormattedDate(item.date_created) }}
+            </td>
+          </template>
+          <template #last_modified="{item}">
+            <td>
+              {{ !!item.last_modified ? getFormattedDate(item.last_modified) : "" }}
+            </td>
+          </template>
+          <template #dealership="{item}" v-if="$store.state.auth.role == 'Administration'">
+            <td>
+              {{ item.dealership.name }}
+            </td>
+          </template>
+        </CDataTable>
+        <CDataTable
+          :items="tableItems"
+          :fields="bulkTableFields"
+          :sorter="true"
+          :hover="true"
+          :clickable-rows="true"
+          :column-filter="true"
+          @row-clicked="selectRow"
+          v-if="selectedBulkAction != 'Bulk Action'"
+        >
+          <template #select="{item}">
+            <td>
+              <CInputCheckbox
+                :checked="item._selected"
+                @update:checked="() => check(item)"
+                custom
+              />
+            </td>
+          </template>
+          <template #date_created="{item}">
+            <td>
+              {{ getFormattedDate(item.date_created) }}
+            </td>
+          </template>
+          <template #last_modified="{item}">
+            <td>
+              {{ !!item.last_modified ? getFormattedDate(item.last_modified) : "" }}
+            </td>
+          </template>
+          <template #dealership="{item}" v-if="$store.state.auth.role == 'Administration'">
+            <td>
+              {{ item.dealership.name }}
+            </td>
+          </template>
+        </CDataTable>
```

Comment:
In fact I don't see anywhere else right now where they can be re-used. However the file is long so refactoring into a separate component like was done in inventory could help the readability of the file

Reply:
A solution to this could be instead of doing an Extract Function refactoring, which would result in just as much code to maintain but in a more readable manner, perhaps a [Combine Functions into Transform](https://refactoring.com/catalog/combineFunctionsIntoTransform.html) refactoring would be better. In other words, combine the 2 tables into one and format it depending on whether we are viewing or deleting.

Reply:
Refactoring [commit](https://github.com/purplesmurf1998/Autotracks/pull/62/commits/98aaa5dad26f6c539ccdc7703658545823c8818a) that fixes the duplication problem. Merged the 2 tables into one and the actions differ depending on the value of the `deletingVehicleList` property